bigtable-hbase-dataflow/src/main/java/com/google/cloud/bigtable/dataflow/CloudBigtableTableConfiguration.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2015 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.bigtable.dataflow;
+
+
+public class CloudBigtableTableConfiguration extends CloudBigtableConfiguration {
+
+  private static final long serialVersionUID = 2435897354284600685L;
+
+  public static CloudBigtableTableConfiguration fromCBTOptions(CloudBigtableOptions options){
+    return new CloudBigtableTableConfiguration(
+        options.getBigtableProject(),
+        options.getBigtableZone(),
+        options.getBigtableClusterId(),
+        options.getBigtableTable());
+  }
+
+  public static class Builder {
+    private String projectId;
+    private String zone;
+    private String cluster;
+    private String table;
+
+    public Builder withProjectId(String projectId) {
```

Comment:
It's the return type that gets you.  Builder is different for all incarnations of withProjectId... it's for chaining purposes.  We can probably just get rid of the Builder concept and use getters/setters.